src/OrchardCore/OrchardCore.Setup.Core/SetupService.cs

```diff
@@ -260,6 +262,19 @@ await scope.ServiceProvider.GetService<IShellDescriptorManager>()
             return executionId;
         }
 
+        // When using SQLite, clearing the connection pool with ReadWriteCreate SqliteOpenMode, used when the shell was
+        // still initializing, to unlock the database file (and thus also allow it to be deleted).
+        if (shellSettings["DatabaseProvider"] == DatabaseProviderValue.Sqlite)
+        {
+            await using var shellContext = await _shellContextFactory.CreateMinimumContextAsync(shellSettings);
+            var store = shellContext.ServiceProvider.GetRequiredService<IStore>();
+            await using var connection = store.Configuration.ConnectionFactory.CreateConnection();
+            if (connection is SqliteConnection sqliteConnection)
+            {
+                SqliteConnection.ClearPool(sqliteConnection);
+            }
```

Comment:
I don't think this code belong here. Maybe in the CreateConntection in the factory? + I don't see you using the connection anywhere

Reply:
I don't see how this can be anywhere else, because we need to clear the pool right before saving the shell settings as Running (changed from `Initializing`).

The connection is used on line 274.

Reply:
You can probably use `ISetupEventHandler`. But, shouldn't we use a wrapper to wrap the SqliteConnection to control how it is disposed?

Reply:
`ISetupEventHandler` can't be used either, because when `SetupAsync()` is called, the tenant is still initializing and any of the event handlers should be able to interact with the DB. The connection pool group can only be cleared when there are no other `ReadWriteCreate` DB connections currently nor will there be any new ones. So, it's too early, and `SucceededAsync()` is too late for that.

This is not about disposing a connection, but disposing a connection pool group. We want to get rid of all the connections in the connection pool for the new tenant's DB, because those lock the file. This is only possible with an explicit `SqliteConnection.ClearPool(sqliteConnection)` call (this uses the `sqliteConnection` to find its group, but the operation is not just for a single connection). Disposing a given connection only returns it to the pool.